Zombidle: Stuck on loading screen

### Describe the bug

[Zombidle](https://www.newgrounds.com/portal/view/671652/format/flash?emulate=flash) gets stuck on the "Preloading stuff..." screen, making the game unplayable.

<img width="967" height="648" alt="Image" src="https://github.com/user-attachments/assets/6644ccc5-61f4-4b5c-8eb5-d5ac2eead057" />

### Expected behavior

The game should load.

### Content Location

Page URL: https://www.newgrounds.com/portal/view/671652/format/flash?emulate=flash
SWF URL: https://uploads.ungrounded.net/671000/671652_zi-fgl.swf

### Affected platform

Desktop app

### Operating system

Windows 10

### Browser

_No response_

### Additional information

The error is:
```
ERROR core/src/avm2/events.rs:411 Error dispatching event EventObject { type: "context3DCreate", class: flash.events::Event, ptr: 0x3072818 } to handler FunctionObject(FunctionObject { ptr: 0x306d1b0, name: "starling.core::Starling/onContextCreated()" }) : ReferenceError: Error #1069: Property uploadFromBitmapData not found on flash.display3D.textures.RectangleTexture and there is no default value.
	at starling.textures::ConcreteTexture/uploadBitmapData()
	at starling.textures::Texture$/fromBitmapData()
	at starling.textures::Texture$/fromBitmap()
	at starling.textures::Texture$/fromEmbeddedAsset()
	at com.lachhhStarling.berzerk::BerzerkEmbeddedBitmapFonts$/loadFonts()
	at com.zombidle::MainGameWebStarling/preloadFirst()
	at Function/http://adobe.com/AS3/2006/builtin::apply()
	at com.lachhh.io::Callback/call()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadNextFla()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadBunch()
	at com.lachhhStarling.berzerk::BerzerkZipModelFlaLoader/loadAllFlas()
	at com.zombidle::MainGameWebStarling/loadModelFlas()
	at com.zombidle::MainGameWebStarling/loadZipFile()
	at Function/http://adobe.com/AS3/2006/builtin::apply()
	at com.lachhh.io::Callback/call()
	at com.lachhhStarling::StarlingMain$/startGame()
	at MethodInfo-5838()
	at starling.utils::AssetManager/loadQueue()
	at com.lachhhStarling::StarlingMain$/loadAssetsDesktop()
	at MethodInfo-5836()
	at starling.events::EventDispatcher/invokeEvent()
	at starling.events::EventDispatcher/dispatchEvent()
	at starling.events::EventDispatcher/dispatchEventWith()
	at starling.core::Starling/initializeRoot()
	at starling.core::Starling/initialize()
	at starling.core::Starling/onContextCreated()
	at flash.events::EventDispatcher/dispatchEventInternal()
	at flash.events::EventDispatcher/dispatchEvent()
	at flash.display::Stage3D/requestContext3D_internal()
	at MethodInfo-2975()
```

--- a/core/src/avm2/globals/flash/display3D/Context3D.as
+++ b/core/src/avm2/globals/flash/display3D/Context3D.as
@@ -27,6 +27,8 @@ package flash.display3D {
         public native function setCulling(triangleFaceToCull:String):void;
         public native function createTexture(width:int, height:int, format:String, optimizeForRenderToTexture:Boolean, streamingLevels:int = 0):Texture;
         public native function createCubeTexture(size:int, format:String, optimizeForRenderToTexture:Boolean, streamingLevels:int = 0):CubeTexture;
+
+        [API("690")]
         public native function createRectangleTexture(width:int, height:int, format:String, optimizeForRenderToTexture:Boolean):RectangleTexture;
 
         public function get driverInfo():String {
@@ -55,6 +57,8 @@ package flash.display3D {
 
         public native function setTextureAt(sampler:int, texture:TextureBase):void;
 
+        // NOTE docs say 694, but playerglobals says 692; playerglobals is correct
+        [API("692")]
         public native function get profile():String;
 
         [API("700")]
@@ -73,6 +77,7 @@ package flash.display3D {
             stub_method("flash.display3D.Context3D", "setStencilReferenceValue");
         }
 
+        [API("686")]
         public native function setSamplerStateAt(sampler:int, wrap:String, filter:String, mipfilter:String):void;
 
         public native function setRenderToTexture(texture:TextureBase, enableDepthAndStencil:Boolean = false, antiAlias:int = 0, surfaceSelector:int = 0, colorOutputIndex:int = 0):void;

